tests/handlers/records-subscribe.spec.ts

```diff
@@ -0,0 +1,773 @@
+          }
+
+          expect(messageCids.length).to.equal(1, 'before delete');
+          expect(messageCids[0]).to.equal(await Message.getCid(chatRecordForBob.message));
+        });
+
+        it('allows $globalRole authorized subscriptions', async () => {
+          // scenario: Alice creates a thread and writes some chat messages writes a chat message. Bob invokes his
+          //           thread member role in order to subscribe to the chat messages.
+
+          const alice = await DidKeyResolver.generate();
+          const bob = await DidKeyResolver.generate();
+
+          const protocolDefinition = friendRoleProtocolDefinition;
+
+          const protocolsConfig = await TestDataGenerator.generateProtocolsConfigure({
+            author: alice,
+            protocolDefinition
+          });
+          const protocolsConfigureReply = await dwn.processMessage(alice.did, protocolsConfig.message);
+          expect(protocolsConfigureReply.status.code).to.equal(202);
+
+          const filter: RecordsFilter = {
+            published    : false,
+            protocol     : protocolDefinition.protocol,
+            protocolPath : 'chat'
+          };
+
+          const noRoleRecords: string[] = [];
+          const addNoRole = async (message: GenericMessage): Promise<void> => {
+            if (message.descriptor.interface === DwnInterfaceName.Records && message.descriptor.method === DwnMethodName.Write) {
+              const recordsWriteMessage = message as RecordsWriteMessage;
+              noRoleRecords.push(recordsWriteMessage.recordId);
+            }
+          };
+
+          // subscribe without role, expect no messages
+          const noRoleSubscription = await TestDataGenerator.generateRecordsSubscribe({
+            author: bob,
+            filter
+          });
+
+          const subscriptionReply = await dwn.processMessage(alice.did, noRoleSubscription.message, { subscriptionHandler: addNoRole });
+          expect(subscriptionReply.status.code).to.equal(200);
+          expect(subscriptionReply.subscription).to.exist;
+
+
+          // Alice writes a 'friend' $globalRole record with Bob as recipient
+          const friendRoleRecord = await TestDataGenerator.generateRecordsWrite({
+            author       : alice,
+            recipient    : bob.did,
+            protocol     : protocolDefinition.protocol,
+            protocolPath : 'friend',
+            data         : new TextEncoder().encode('Bob is my friend'),
+          });
+          const friendRoleReply = await dwn.processMessage(alice.did, friendRoleRecord.message, { dataStream: friendRoleRecord.dataStream });
+          expect(friendRoleReply.status.code).to.equal(202);
+
+          const recordIds: string[] = [];
+          const addRecord = async (message: GenericMessage): Promise<void> => {
+            if (message.descriptor.interface === DwnInterfaceName.Records && message.descriptor.method === DwnMethodName.Write) {
```

Comment:
Consider:

```suggestion
          const addRecord: RecordSubscriptionHandler = async (message): Promise<void> => {
            if (message.descriptor.method === DwnMethodName.Write) {
```

Reply:
Fixed here and another identical place in the tests.